### PR TITLE
[automation/dotnet] Expose WorkspaceStack.GetOutputsAsync 

### PR DESCRIPTION
**Pull request by vipentti**
Other automation API's allow for getting stack outputs separately.
This makes `GetOutputAsync` public so outputs can be accessed
outside of running `Up`.

**NOTE** I added similar stack output tests to all automation APIs for consistency and because they already supported the expected behavior no changes to the APIs themselves were necessary.

### CODE REVIEW

File: sdk/dotnet/Pulumi.Automation/WorkspaceStack.cs
Comment (vipentti) [inline sdk/dotnet/Pulumi.Automation/WorkspaceStack.cs:533]: Do we want to keep the singular name `GetOutputAsync` or use plural `GetOutputsAsync`? Other APIs have `outputs`

Comment (komalali) [reply sdk/dotnet/Pulumi.Automation/WorkspaceStack.cs:533]: Let's change it to `GetOutputs` to align with the other SDKs and more accurately reflect the functionality.

**Review by komalali — APPROVED** [review]
The changes look good, I'm just waiting for the tests to pass before I merge. Ran into a couple of flakey network issues in the last few runs.

**Pull request merged**
